### PR TITLE
kola/tests/etcd: make etcd1Discovery pass

### PR DESCRIPTION
**Pull request by pbx0**
Check health by setting and getting a few keys.

### CODE REVIEW

File: kola/tests/etcd/discovery.go [hunk 1]
```diff
@@ -62,9 +64,41 @@ func discovery(cluster platform.Cluster, version int) error {
...
+		}
+	} else if version == 1 {
+		var keyMap map[string]string
+		var retryFuncs []func() error
+
+		retryFuncs = append(retryFuncs, func() error {
+			var err error
+			keyMap, err = setKeys(cluster, 5)
+			if err != nil {
+				//return err
```
Comment (mischief) [inline kola/tests/etcd/discovery.go]: get rid of the commented returns

File: kola/tests/etcd/discovery.go [hunk 2]
```diff
@@ -62,9 +64,41 @@ func discovery(cluster platform.Cluster, version int) error {
...
+		retryFuncs = append(retryFuncs, func() error {
+			if err := checkKeys(cluster, keyMap, false); err != nil {
+				//return err
+				return fmt.Errorf("checkkeys %v", err)
+			}
+			return nil
+		})
+		for _, retry := range retryFuncs {
+			if err := util.Retry(5, 5*time.Second, retry); err != nil {
+				return err
```
Comment (mischief) [inline kola/tests/etcd/discovery.go]: might as well wrap this in the same error message that etcd2 discovery check would print for consistency:

`return fmt.Errorf("discovery failed health check: %v", err)`

File: kola/tests/etcd/discovery.go (hunk 1 above)
Comment (pbx0) [inline kola/tests/etcd/discovery.go]: done -- good catch those were left from debugging

File: kola/tests/etcd/discovery.go (hunk 2 above)
Comment (pbx0) [inline kola/tests/etcd/discovery.go]: done

**Comment by mischief** [comment]
pending build passing on jenkins, lgtm

**Comment by mischief** [comment]
not lgtm. this can't work as is since you exec curl on the machine running the tests, not on the coreos machines. this won't work on aws where the only open port on the public ip is ssh.

you need to rework the key-checking to run curl over the ssh tunnel to the machines.

**Comment by pbx0** [comment]
@michief PTAL

The jenkins build fail seems to be unrelated, I have all tests passing locally.

**Comment by mischief** [comment]
retest this please

**Comment by mischief** [comment]
lgtm

**Pull request merged**